mitiq/executor/executor.py

```diff
@@ -44,13 +47,17 @@
     List[float],
     Sequence[float],
     Tuple[float],
+    list[float],
+    tuple[float],
 ]
 MeasurementResultLike = [
     MeasurementResult,
     Iterable[MeasurementResult],
```

Comment:
Doesn't this line make all other iterables (list, tuple, etc.) redundant?

Reply:
No, because of the way we use `MeasurementResultLike`. E.g.

https://github.com/unitaryfund/mitiq/blob/7e9641014f5e27a15942775c99067a94e88a5969/mitiq/executor/executor.py#L174

If we did something like checking if the return type is a subclass then perhaps this would work, but then we would need to have different checks for batched/not batched executors.
```py
if any(issubclass(self._executor_return_type, Iterable[MeasurementResult])):
    ...  # I haven't tried this
```